src/DSi_TMD.h

```diff
@@ -0,0 +1,157 @@
+/*
+    Copyright 2016-2023 melonDS team
+
+    This file is part of melonDS.
+
+    melonDS is free software: you can redistribute it and/or modify it under
+    the terms of the GNU General Public License as published by the Free
+    Software Foundation, either version 3 of the License, or (at your option)
+    any later version.
+
+    melonDS is distributed in the hope that it will be useful, but WITHOUT ANY
+    WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+    FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with melonDS. If not, see http://www.gnu.org/licenses/.
+*/
+
+#ifndef DSI_TMD_H
+#define DSI_TMD_H
+
+#include "types.h"
+#include <array>
+
+namespace DSi_TMD
+{
+
+struct [[gnu::packed]] TitleMetadataContent {
+    /// Content ID (00,00,00,vv) ;lowercase/hex ;"0000000vv.app"
+    u8 ContentId[4];
+
+    /// Content Index (00,00)
+    u16 ContentIndex;
+
+    /// Content Type (00,01) ;aka DSi .app
+    u16 ContentType;
+
+    /// Content Size (00,00,00,00,00,19,E4,00)
+    u64 ContentSize;
+
+    /// Content SHA1 (on decrypted ".app" file)
+    u8 ContentSha1Hash[0x14];
+
+    [[nodiscard]] u32 GetVersion() const noexcept
+    {
+        return (ContentId[0] << 24) | (ContentId[1] << 16) | (ContentId[2] << 8) | ContentId[3];
+    }
+};
+
```

Comment:
Might be good to add a static assert for this struct as well.

Reply:
I figured that the `static_assert` for the overall `TitleMetadata` would suffice, but sure. Why not?